packages/renderer/src/stores/containers.ts

```diff
@@ -65,7 +65,21 @@ const containersEventStore = new EventStore<ContainerInfo[]>(
 containersEventStore.setupWithDebounce();
 
 export const searchPattern = writable('');
+export const runningFilter = writable(false);
+export const stoppedFilter = writable(false);
 
-export const filtered = derived([searchPattern, containersInfos], ([$searchPattern, $containersInfos]) =>
-  $containersInfos.filter(containerInfo => findMatchInLeaves(containerInfo, $searchPattern.toLowerCase())),
+export const filtered = derived(
+  [searchPattern, runningFilter, stoppedFilter, containersInfos],
+  ([$searchPattern, $runningFilter, $stoppedFilter, $containersInfos]) =>
+    $containersInfos
+      .filter(containerInfo => findMatchInLeaves(containerInfo, $searchPattern.toLowerCase()))
+      .filter(containerInfo => {
+        if ($runningFilter) {
+          return containerInfo.State.toUpperCase() === 'RUNNING';
+        }
+        if ($stoppedFilter) {
+          return containerInfo.State.toUpperCase() !== 'RUNNING';
+        }
+        return true;
+      }),
```

Comment:
to me it's strange that the state of the filter is in the store ?
shouldn't it belong to container list display thing ?

Reply:
Is it different from the searchPattern, which belongs to the state?

Reply:
hello @feloy, you're correct no it's not so different from the search pattern but I'm not sure search pattern is at the right place as well ?!

I mean, do we consider the stores as the only data being exposed in all the UIs of Podman Desktop or it's just a representation state of the current state on the server

for example, is that if I order by Age in container list I want that in another view containers will be sorted in the same way or it depends on the view

so, is that filtering is part of the store or the view

it would be ok for me if there is a allContainers (what we have) a runningContainers and a stoppedContainers objects
but do we want to display all running containers in more than one location ?

Reply:
Yes, I also think that the state should contain the representation of what exists in the machine, and this information should be transformed depending on where it is displayed.

For the moment, I don't think the list of containers is displayed in several places, but I'm pretty sure we will want to add some counters somewhere in the UI, and the state won't be helpful for that.

I can look for moving the searchPattern and the running state out of the state.

Reply:
you could also for the running/all/stopped, have a filter that is being applied on the route

all : noop filter, running = filter out what is not running, stopped = keep what is not running

Reply:
> you could also for the running/all/stopped, have a filter that is being applied on the route
> 
> all : noop filter, running = filter out what is not running, stopped = keep what is not running

Hi @benoitf sorry I don't understand completely what you mean. Could you elaborate please?

Reply:
I mean that we could have different fields in the store where it represents different view (not saying that we must)

For now we have a field with all containers (containerInfos)

But we could have in the container store other objects representing different states like all the running containers and another one with all non-running containers.

Sso it's still accurate data, it does not depends on a user filter, etc

Reply:
ok, got it. It would be interesting if these infos were used in different places. I would prefer to have them located in the component for now, and move them to the state when we use it in different places.